client/src/sections/courses/LearningSection/AddLearnerModal.tsx

```diff
@@ -1,125 +1,177 @@
+/* eslint-disable multiline-ternary */
+import React, { useState } from 'react';
+import { useGetEnrolleesQuery } from '@/services/enrolleeAPI';
 import Button from '@/src/shared/components/Button';
 import Modal from '@/src/shared/components/Modal/Modal';
 import Pagination from '@/src/shared/components/Pagination';
 import SearchBar from '@/src/shared/components/SearchBar/SearchBar';
 import CloseIcon from '@/src/shared/icons/CloseIcon';
-import { type Learner } from '@/src/shared/utils';
-import React, { useState } from 'react';
-
-interface AddLearnerModalProps {
-  learners: Learner[];
-}
+import { useRouter } from 'next/router';
+import { useDispatch } from 'react-redux';
+import type { AppDispatch } from '@/app/store';
+import { storeEnrollee } from '@/services/posts/addNewEnrollees';
 
-const AddLearnerModal: React.FC<AddLearnerModalProps> = ({ learners }: AddLearnerModalProps): JSX.Element => {
+const AddLearnerModal: React.FC = (): JSX.Element => {
+  const router = useRouter();
+  const params = router.query;
   const [currentPage, setCurrentPage] = useState(1);
   const [isModalOpen, setIsModalOpen] = useState(false);
+  const [search, setSearch] = useState('');
+  const [page, setPage] = useState(1);
+  const [selectedIds, setSelectedIds] = useState<number[]>([]);
+  const courseID = params.id;
+  const dispatch = useDispatch<AppDispatch>();
+  const { data, refetch } = useGetEnrolleesQuery({
+    courseID,
+    search,
+    page,
+  });
 
-  const handleAddLearners = (): void => {
-    alert('Add Learners');
-  };
+  const enrollees = data?.enrollees;
 
   const handleModalClose = (): void => {
+    setSelectedIds([]);
     setIsModalOpen(false);
+    setSearch('');
+    setPage(1);
+    void refetch();
   };
 
-  const handleSearch = (search: string): void => {
-    alert(`You searched ${search}`);
+  const handleAddMaterialModal = (): void => {
+    setSelectedIds([]);
+    setSearch('');
+    setPage(1);
```

Comment:
I'm assuming its for opening the add learner modal, and if its:

If you need the function, please rename the variable to `handleOpenModal`, but I dont think you do, because you can remove all     
```
   setSelectedIds([]);
    setSearch('');
    setPage(1);
```
since they are already been resetted in `handleModalClose`, so you do not need to reset them again when opening the modal